websites/H/HBO Max/presence.ts

```diff
@@ -65,30 +67,67 @@ function getToken(): Promise<string> {
 					paymentProviderCode: "blackmarket"
 				}
 			})
+			/* eslint-enable camelcase */
 		})
-			.then(x => x.json())
-			.then(x => resolve(x.access_token));
-	});
+	).json();
+
+	return response.access_token;
 }
-/* eslint-enable camelcase */
 
-async function fetchCover() {
+async function fetchClientConfig(token: string) {
 	const response = await (
-		await fetch(
-			`https://comet.api.hbo.com/express-content/${
-				location.pathname.split("/")[2]
-			}?device-code=desktop&product-code=hboMax&api-version=v9.0&country-code=US&language=en-us`,
-			{
-				headers: {
-					authorization: `Bearer ${await getToken()}`
-				}
-			}
-		)
+		await fetch("https://sessions.api.hbo.com/sessions/v1/clientConfig", {
+			method: "POST",
+			headers: {
+				authorization: `Bearer ${token}`,
+				"content-type": "application/json"
+			},
+			body: JSON.stringify({
+				contract: "abc:1.0.0.0",
+				preferredLanguages: ["en-us"]
+			})
+		})
 	).json();
 
-	return `https://artist.api.cdn.hbo.com/images/${
-		response[0].body.references.series.match(/series:([^:]+)/)[1]
-	}/tileburnedin?size=1024x1024`;
+	return [
+		response.routeKeys.contentSubdomain,
+		new URLSearchParams(
+			response.features["express-content"].config.expressContentParams
+		).get("country-code")
+	];
+}
+
+async function fetchCover() {
+	let output: string;
+
+	const accessToken = await fetchToken(),
+		[routeKey, countryCode] = await fetchClientConfig(accessToken);
+
+	isFetching = true;
+
+	try {
+		const response = await (
+			await fetch(
+				`https://comet${routeKey}.api.hbo.com/express-content/${
+					location.pathname.split("/")[2]
+				}?device-code=desktop&product-code=hboMax&api-version=v9.0&country-code=${countryCode}&language=en-us`,
+				{
+					headers: {
+						authorization: `Bearer ${accessToken}`
+					}
+				}
+			)
+		).json();
+
+		output = `https://artist.api.cdn.hbo.com/images/${
+			response[0].body.references.series.match(/series:([^:]+)/)[1]
+		}/tileburnedin?size=1024x1024`;
+	} catch {
+		output = "lg";
```

Comment:
just curious, why lg?

Reply:
Are you asking why it's named lg? If so I don't know. I don't own the application

Reply:
I'm asking why you're returning "lg" when an error is found if otherwise you're returning a URL

Reply:
I just thought it would be fine just to return the logo as a fallback. Do I have to use .catch or what

Reply:
Nono I just wanted to understand why you used lg. is that the key for the logo in the presence assets?

Reply:
Yes

Reply:
Alright then